packages/stream_chat_flutter/lib/src/message_list_view/floating_date_divider.dart

```diff
@@ -47,28 +49,37 @@ class FloatingDateDivider extends StatelessWidget {
           return const Empty();
         }
 
-        var index = switch (reverse) {
+        final index = switch (reverse) {
           true => getBottomElementIndex(positions),
           false => getTopElementIndex(positions),
         };
 
-        if ((index == null) ||
-            (!isThreadConversation && index == itemCount - 2) ||
-            (isThreadConversation && index == itemCount - 1)) {
-          return const Empty();
-        }
+        if (index == null) return const Empty();
+
+        // We can only calculate the date divider if the element is a message
+        // widget and not one of the special items.
+
+        // Parent Message
+        if (index == itemCount - 1) return const Empty();
+        // Header Builder
+        if (index == itemCount - 2) return const Empty();
+        // Top Loader Builder
+        if (index == itemCount - 3) return const Empty();
+        // Bottom Loader Builder
+        if (index == 1) return const Empty();
+        // Footer Builder
+        if (index == 0) return const Empty();
```

Comment:
I would create a different method for it, this is adding quite some noise

Just like this:
```dart
if(!_shouldShowDivider(index)) return const Empty();
```

with:
```dart
bool _shouldShowDivider(int index){
  return index != null &&
    
    // We can only calculate the date divider if the element is a message
    // widget and not one of the special items.

    // Parent Message
    index != itemCount - 1 &&
    // Header Builder
    index != itemCount - 2 &&
    // Top Loader Builder
    index != itemCount - 3 &&
    // Bottom Loader Builder
    index != 1 &&
    // Footer Builder
    index != 0;
}